receiver/githubreceiver/internal/scraper/githubscraper/helpers.go

```diff
@@ -211,6 +211,99 @@ func (ghs *githubScraper) getPullRequests(
 	return pullRequests, nil
 }
 
+// getMergedPullRequests fetches merged pull requests with optional time-based filtering.
+// Uses backward pagination (last/before) to efficiently stop when hitting the cutoff date.
+// If lookbackDays is 0, fetches all merged PRs. Otherwise, only fetches PRs merged
+// within the last N days.
+func (ghs *githubScraper) getMergedPullRequests(
+	ctx context.Context,
+	client graphql.Client,
+	repoName string,
+	lookbackDays int,
+) ([]MergedPullRequestNode, error) {
+	var cursor *string
+	var pullRequests []MergedPullRequestNode
+
+	// Calculate cutoff time if lookback is enabled (non-zero)
+	var cutoffTime time.Time
+	if lookbackDays > 0 {
```

Comment:
Just a suggestion, we're using the check `lookbackDays > 0` a couple of times here to check if the user wants to limit how many days to lookback. We could instead use a new bool here and simply name it something like `limitLookbackDays`. Maybe this would easier to follow in future checks vs. the reader having to remember the context of why we care about lookbackDays being set to a positive value.

I'll leave it up to you though, it's a tiny nit.